examples/metadata_processing/plot_add_dict_entries.py

```diff
@@ -1,49 +1,75 @@
 """
-
-# authors : Darcy Mason
-#           Guillaume Lemaitre <g.lemaitre58@gmail.com>
+# authors : Darcy Mason and pydicom contributors
 # license : MIT
 
 
-from pydicom.datadict import DicomDictionary, keyword_dict
+from pydicom import dcmread
+from pydicom.datadict import add_private_dict_entries
 from pydicom.dataset import Dataset
+from pydicom.filebase import DicomBytesIO
+from pydicom.valuerep import VR
 
 print(__doc__)
 
-# Define items as (VR, VM, description, is_retired flag, keyword)
-#   Leave is_retired flag blank.
+# create a dataset with some private tags for demonstration
+# we create the dataset with Implicit VR Little Endian transfer syntax,
+# so that the VR of the private tags will not be saved
+
+ds = Dataset()
+ds.is_implicit_VR = True
+ds.is_little_endian = True
+
+# add private tags by creating a new private block and add elements to it
+block = ds.private_block(0x1001, "ACME 3.1", create=True)
+block.add_new(0x01, VR.UL, 42)
+block.add_new(0x02, VR.SH, "Hello World")
+block.add_new(0x03, VR.UI, "1.2.3.4.5")
+
+# write the dataset into a memory file and read it back
+# this simulates reading from a normal DICOM file
+fp = DicomBytesIO()
+ds.save_as(fp)
+ds = dcmread(fp, force=True)
+
+print("Output for unknown private tags:")
+print(ds)
+
+# Creates output:
+# (1001,0010) Private Creator                     VR.LO: 'ACME 3.1'
+# (1001,1001) Private tag data                    VR.UN: b'*\x00\x00\x00'
+# (1001,1002) Private tag data                    VR.UN: b'Hello World '
+# (1001,1003) Private tag data                    VR.UN: b'1.2.3.4.5\x00'
+
+
+# Add the private tags to the private tag dictionary
+
+# Define items as (VR, VM, description, is_retired flag)
+# Leave is_retired flag blank.
 new_dict_items = {
-    0x10011001: ("UL", "1", "Test One", "", "TestOne"),
-    0x10011002: ("OB", "1", "Test Two", "", "TestTwo"),
-    0x10011003: ("UI", "1", "Test Three", "", "TestThree"),
+    0x10011001: ("UL", "1", "Test One", ""),
+    0x10011002: ("SH", "1", "Test Two", ""),
+    0x10011003: ("UI", "1", "Test Three", ""),
 }
 
-# Update the dictionary itself
-DicomDictionary.update(new_dict_items)
-
-# Update the reverse mapping from name to tag
-new_names_dict = dict([(val[4], tag) for tag, val in new_dict_items.items()])
-keyword_dict.update(new_names_dict)
+# add the entries to the private dictionary, using the correct private creator string
+add_private_dict_entries(private_creator="ACME 3.1", new_entries_dict=new_dict_items)
 
-# Test that it is working
-ds = Dataset()  # or could get one from dcmread, etc
+# re-read the dataset for the new dictionary entries to be applied
+ds = dcmread(fp, force=True)
 
-ds.TestOne = 42
-ds.TestTwo = "12345"
-ds.TestThree = "1.2.3.4.5"
+print("\nOutput with registered private tags:")
+print(ds)
 
-print(ds.top())
+# Creates output:
+# (1001,0010) Private Creator                     VR.LO: 'ACME 3.1'
+# (1001,1001) [Test One]                          UL: 42
```

Comment:
The display of the VR (`VR.LO` vs `UL`) is currently that way - this is probablt a bug, haven't looked at it yet.

Reply:
There [was an issue](https://github.com/pydicom/pydicom/issues/1874) that showed something similar, but I couldn't reproduce it

Reply:
Argh, yeah, [its a Python issue](https://github.com/python/cpython/issues/100458). With 3.11 I get:
```python
>>> from pydicom.valuerep import VR
>>> f"{VR.LO"}
'VR.LO'
```
Will need to add version dependent switch to [StrEnum](https://docs.python.org/3/library/enum.html#enum.StrEnum) for v3.11+

I can take care of that

Reply:
Thanks for the fix - I have changed the output accordingly now.